File: src/main/kotlin/com/dokhabackend/dokha/config/security/JwtAuthenticationFilter.java
package com.dokhabackend.dokha.config.security;

import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
import org.springframework.web.filter.OncePerRequestFilter;

import javax.security.sasl.AuthenticationException;
import javax.servlet.FilterChain;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

/**
 * @author v.butuzov
 * Фильтр авторизации через jwt
 */
@Slf4j
public class JwtAuthenticationFilter extends OncePerRequestFilter {

    @Value("${jwt.header}")
    private String HEADER_STRING;

    @Qualifier("userServiceImpl")
    @Autowired
    private UserDetailsService userDetailsService;

    @Autowired
    private JwtTokenUtil jwtTokenUtil;

    @Override
    protected void doFilterInternal(HttpServletRequest req, HttpServletResponse res, FilterChain chain) throws IOException, ServletException {
        String header = req.getHeader(HEADER_STRING);
        String uri = req.getRequestURI();
        String login;

        try {
            if (uri.contains("register") || uri.contains("login") || uri.contains("/v2/api-docs")) {

            } else {
                if (header == null || header.isEmpty()) {
                    throw new AuthenticationException("Пользователь не авторизован");
                }

                login = jwtTokenUtil.getUsernameFromToken(header);

                if (login != null && SecurityContextHolder.getContext().getAuthentication() == null) {

                    UserDetails userDetails = userDetailsService.loadUserByUsername(login);

                    if (jwtTokenUtil.validateToken(header, userDetails)) {
                        UsernamePasswordAuthenticationToken authentication = new UsernamePasswordAuthenticationToken(userDetails, null, userDetails.getAuthorities());
                        authentication.setDetails(new WebAuthenticationDetailsSource().buildDetails(req));
                        logger.info("Вход выполнен. Пользователь: " + login);
                        SecurityContextHolder.getContext().setAuthentication(authentication);
                    }
                }
            }
        } catch (Exception e) {
            log.error("Несуществующий логин", e);
            throw new IllegalArgumentException("Пользователь не авторизован");
        }

        chain.doFilter(req, res);
    }
}

File: src/main/kotlin/com/dokhabackend/dokha/config/security/JwtAuthenticationFilter.kt
package com.dokhabackend.dokha.config.security

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.security.authentication.RememberMeAuthenticationToken
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.stereotype.Component
import org.springframework.web.filter.OncePerRequestFilter
import javax.security.sasl.AuthenticationException
import javax.servlet.FilterChain
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

/**
 * @author v.butuzov
 * Фильтр авторизации через jwt
 */

@Component
class JwtAuthenticationFilter : OncePerRequestFilter() {

    @Qualifier("userServiceImpl")
    @Autowired
    private lateinit var userDetailsService: UserDetailsService

    @Autowired
    private lateinit var jwtProperties: JwtProperties

    override fun doFilterInternal(req: HttpServletRequest, res: HttpServletResponse, chain: FilterChain) {
        val token = req.getHeader(jwtProperties.header)
        val uri = req.requestURI

        if (uri.contains("/register") || uri.contains("/login"))

        else if (token == null || token.isEmpty())
            throw AuthenticationException("Вася а где токен?")
        else {
            val login = JwtTokenUtil(jwtProperties).getUsernameFromToken(token)
            val user = userDetailsService.loadUserByUsername(login)

            SecurityContextHolder.getContext().authentication = RememberMeAuthenticationToken(token, user, user.authorities)
        }
        chain.doFilter(req, res)
    }
}